cabal-install/Distribution/Client/ProjectPlanning/Types.hs

```diff
@@ -428,8 +427,10 @@ elabExeDependencyPaths elab =
 elabSetupDependencies :: ElaboratedConfiguredPackage -> [ConfiguredId]
 elabSetupDependencies elab =
     case elabPkgOrComp elab of
-        ElabPackage pkg    -> CD.setupDeps (pkgLibDependencies pkg)
-        ElabComponent comp -> compSetupDependencies comp
+        ElabPackage pkg -> CD.setupDeps (pkgLibDependencies pkg)
+        -- TODO: Custom setups not supported for components yet.  When
```

Comment:
Yes, but I think this chunk of code needs to be rewritten in that regime which is why I put the TODO in.